instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java

```diff
@@ -65,10 +68,32 @@ public JavaHttpClientTelemetryBuilder setCapturedResponseHeaders(List<String> re
     return this;
   }
 
+  /**
+   * Configures the instrumentation to recognize an alternative set of HTTP request methods.
+   *
+   * <p>By default, this instrumentation defines "known" methods as the ones listed in <a
+   * href="https://www.rfc-editor.org/rfc/rfc9110.html#name-methods">RFC9110</a> and the PATCH
+   * method defined in <a href="https://www.rfc-editor.org/rfc/rfc5789.html">RFC5789</a>.
+   *
+   * <p>Note: calling this method <b>overrides</b> the default known method sets completely; it does
+   * not supplement it.
+   *
+   * @param knownMethods A set of recognized HTTP request methods.
+   */
+  @CanIgnoreReturnValue
+  public JavaHttpClientTelemetryBuilder setKnownMethods(Set<String> knownMethods) {
+    this.knownMethods = knownMethods;
```

Comment:
Nit: Wrap with a new set so the caller can't sneaky mutate later. Arguably overkill for a builder.

Reply:
This is gonna go away when I implement https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8779#discussion_r1237096199